controllers/operands/kubevirt.go

```diff
@@ -802,16 +803,24 @@ func NewKubeVirtWithNameOnly(hc *hcov1beta1.HyperConverged, opts ...string) *kub
 	}
 }
 
-func hcoConfig2KvConfig(hcoConfig hcov1beta1.HyperConvergedConfig, infrastructureHighlyAvailable bool) *kubevirtcorev1.ComponentConfig {
-	if hcoConfig.NodePlacement == nil && infrastructureHighlyAvailable {
+func hcoConfig2KvConfig(hcoConfig hcov1beta1.HyperConvergedConfig, controlPlaneHighlyAvailable, controlPlaneNodeExists bool) *kubevirtcorev1.ComponentConfig {
+	if hcoConfig.NodePlacement == nil && controlPlaneHighlyAvailable {
 		return nil
 	}
 
 	kvConfig := &kubevirtcorev1.ComponentConfig{}
-	if !infrastructureHighlyAvailable {
+	if !controlPlaneHighlyAvailable {
 		kvConfig.Replicas = ptr.To[uint8](1)
 	}
 
+	// In case there are no worker nodes, we're setting an empty struct for NodePlacement
```

Comment:
Comments are a code smell. In this case, the English says one thing, and the code says another.
It is better to have no comment. When you feel that comments are required, it is typically a sign that you should create a small peroperly-named helper function that should be used here. e.g `func kvNodePlacement(hcoConfig, controlPlaneNodeExists)`

Reply:
Since it wasn't so obvious, I felt it's reasonable to add a bit explanation for the rationale behind this condition, so future code readers will know exactly why we have this logic.

Reply:
I think that future readers are no worse off, since the English comment is wrong.

Reply:
oh, I now got what you meant. Thanks for the catch. Fixed the comment in #3231